### PR TITLE
Weekly #6: March 1st-7th

### PR DESCRIPTION
**Pull request by RichardLitt**

### CODE REVIEW

**Comment by daviddias** [comment]
To be added:
- [x] New libp2p-spdy and libp2p-multiplex for all your Stream Multiplexing needs
  - https://github.com/diasdavid/js-libp2p-spdy
  - https://github.com/diasdavid/js-libp2p-multiplex
- [x] Do you want to become an JS IPFS contributor? Check out the captain.log https://github.com/ipfs/js-ipfs/issues/30#issuecomment-187950929
- [x] libp2p-swarm is getting a lot of work done, new API, more tests, less magic, more flexibility and awesomeness https://github.com/diasdavid/js-libp2p-swarm/pull/20

More info at - https://github.com/ipfs/pm/issues/93#issuecomment-193301659

**Comment by daviddias** [comment]
To be added:

[station](https://github.com/ipfs/station) - The new version of `station` is ready for developer preview! `station` is one of the easiest ways to get an IPFS daemon running it your machine, it acts as a service and you get all the convenient features such as: Turn a IPFS node on through a GUI and drag and drop to share through IPFS. To try it out, you need Node.js 4 and NPM 3 installed, then do the following:

```
> git clone https://github.com/ipfs/station.git && cd station
> npm install
> npm start
```

image

@RichardLitt

File: published/006-2016-March-7.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  agnostic, message gossip protocol. Any peer in the network can publish
+  messages, which will eventually propagate to the entire network via rounds of
+  gossip between each node's peers.
+- [pubsub-swarm](https://github.com/noffle/pubsub-swarm) - Form a p2p swarm of
+  nodes around a topic and exchange messages, with a traditional pubsub API.
+
+#### [**go-ipfs**](//github.com/ipfs/go-ipfs)
+
+@whyrusleeping set up [Teamcity](https://www.jetbrains.com/teamcity/). This cuts down on the long waits for Travis to run, and hopefully will mean faster CI tests. 
+Teamcity also gives us awesome metrics on our tests and nice statistics on failures and failure rates. Weigh in on this change by joining the discussion, [here](TODO).
```
Comment (whyrusleeping) [inline published/006-2016-March-7.md]: can also add that there is now a `Default` method that can be used when creating commands for cleaner command code and nicer autogenerated documentation.

**Comment by ghost** [comment]
This LGTM! :+1: :ship:

File: published/006-2016-March-7.md (hunk 1 above)
Comment (whyrusleeping) [inline published/006-2016-March-7.md]: Can also add to the teamcity part, something like:

"teamcity also has nice integration with a large array of test runners, from go tests to karma and sharness. It will give us more detailed feedback about our test runs, even when successful"

Comment (RichardLitt) [inline published/006-2016-March-7.md]: I'll add the default in the next weekly, after I overhaul the current docs.

**Comment by RichardLitt** [comment]
To do:
- [x] @eminence Add in contributors from commits.
- [ ] @jbenet CR

**Comment by eminence** [comment]
The current set of contributors from commits is an exact subset of the contributors you've already included, so that's all set!

**Comment by RichardLitt** [comment]
@eminence Awesome! Thanks!

**Comment by RichardLitt** [comment]
To do: 
- [x] Comb Twitter for community things.

**Comment by ghost** [comment]
> Comb Twitter for community things.

http://www.sitepoint.com/http-vs-ipfs-is-peer-to-peer-sharing-the-future-of-the-web/

File: published/006-2016-March-7.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+#### [**randor**](http:https://github.com/dignifiedquire/randor)
+
+[@dignifiedquire](https://github.com/dignifiedquire) has been working on [randor](http:https://github.com/dignifiedquire/randor), a testing framework that will be able to send huge files and lots of requests into IPFS in order to test how it works for edge cases and how it scales. Randor is now able to rerun tests predictably based on stored data, so it's easy to find and fix bugs. [@whyrusleeping](https://github.com/whyrusleeping) is already working on the first bug that randor detected. To contribute, check out the repository.
+
+#### [**WebRTC**](https://github.com/diasdavid/webrtc-explorer)
```
Comment (hackergrrl) [inline published/006-2016-March-7.md]: s/WebRTC/WebRTC Explorer/ ?

Comment (daviddias) [inline published/006-2016-March-7.md]: makes sense, I missed that. Thank you @noffle :)

File: published/006-2016-March-7.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+If you would like to get this update as an email, sign up for our [weekly newsletter](https://tinyletter.com/ipfsweekly)!
+
+Here are some of the highlights for the [first week of March](https://github.com/ipfs/pm/issues/93):
+
+## Updates
+
+#### [**station**](https://github.com/ipfs/station)
+
+![image](https://cloud.githubusercontent.com/assets/1211152/13579734/e5dd5c96-e496-11e5-99cf-f3c09943f2f4.png)
```
Comment (jbenet) [inline published/006-2016-March-7.md]: all weekly images should be added to the same folder the post is in, and hosted out of IPFS

Comment (jbenet) [inline published/006-2016-March-7.md]: @dignifiedquire can we get an image with more peers? like 100+? 3 just looks sad and lonely.

File: published/006-2016-March-7.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+The new version of `station` is ready for developer preview! `station` is one of the easiest ways to get an IPFS daemon running on your machine. It acts as a service and you get many convenient features, such as the ability to turn an IPFS node on through a GUI and drag and drop to share through IPFS. To try it out, you need Node.js 4 and npm 3 installed. Then, do the following:
```
Comment (jbenet) [inline published/006-2016-March-7.md]: suggestion: make the first "station" here a link. the image breaks up, and the link is hard to see.

Comment (jbenet) [inline published/006-2016-March-7.md]: add a link to install instructions to nodejs 4 and npm 3 (like "(see install instructions here)")

File: published/006-2016-March-7.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+> npm start
+```
+
+#### [**website**](https://github.com/ipfs/website)
+
+The list of [API Commands](https://ipfs.io/docs/commands/) on the website has been updated. This provides a single place to look through all of the CLI commands for go-ipfs, at once; it is a good reference point if you're not sure exactly which command to use next, and `ipfs commands` seems sparse.
+
+#### **AUR**
+
+[@Kubuxu](https://github.com/Kubuxu) has made sure that `gx`, `gx-go`, and `ipget` are available on [AUR](https://aur.archlinux.org), the arch-linux user repository. This means that it will be easier to get and install these on linux platforms.
```
Comment (jbenet) [inline published/006-2016-March-7.md]: "on some linux platforms"

File: published/006-2016-March-7.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+#### [**WebRTC Explorer**](https://github.com/diasdavid/webrtc-explorer)
+
+WebRTC Explorer 2.0.0 has been alpha released! WebRTC Explorer is essentially a way to bring Packet Switching to the Application Level, using WebRTC Data Channels as the transportation layer between nodes. WebRTC enables communication between browsers without needing mediators (servers), enabling users to route packets between machines, using only Web technologies. WebRTC is inspired by the Chord routing scheme to create routing tables that are evenly balanced across nodes. [@diasdavid](https://github.com/diasdavid) wrote up a more extensive [blog post here](http://blog.daviddias.me/2016/02/28/webrtc-explorer-2-0-0-alpha-release), and has an introductory video [here](https://youtu.be/fNQGGGE__zI).
```
Comment (jbenet) [inline published/006-2016-March-7.md]: - @diasdavid i think "WebRTC Explorer" is more than "a way to bring Packet Switching to the Application Level" no?
- "WebRTC is inspired by the Chord routing scheme to create..." you mean "WebRTC Explorer". WebRTC alone has nothing to do with Chord.

**Comment by jbenet** [comment]
@RichardLitt some comments above, otherwise LGTM! ready to rock and roll.

Really like the style of this one, was an exciting read :) :+1:

**Comment by jbenet** [comment]
@RichardLitt maybe add one more:

> ##### Science Data
> 
> @jbenet visited the [Janelia Research Campus](https://www.janelia.org) to learn about scientific data tools and use cases. He gave a talk about IPFS, data versioning, package management, and more (video forthcoming). He learned about [DVID](https://github.com/janelia-flyem/dvid), and the requirements of the amazing [FlyEM brain imaging effort](https://www.janelia.org/project-team/flyem). He got to see many fantastic open source research tools made by the [FlyEM Team](https://www.janelia.org/project-team/flyem) ([github](https://github.com/janelia-flyem)), the [Freeman Lab](http://thefreemanlab.com/) ([github](https://github.com/freeman-lab)), and other groups. Go check out their github repos, and help them improve brain research!

**Comment by hackergrrl** [comment]
+1 for format. I find this much easier to mentally process.

File: published/006-2016-March-7.md (hunk 3 above)
Comment (RichardLitt) [inline published/006-2016-March-7.md]: @jbenet I don't understand. I can copy them to the current directory, and link to them (on the Blog) and embed them manually (for TinyLetter, GitHub). That seems unrelated to hosting them out of IPFS.

**Comment by RichardLitt** [comment]
@jbenet I'm going to add this file to. Sound good?

File: published/006-2016-March-7.md (hunk 3 above)
Comment (dignifiedquire) [inline published/006-2016-March-7.md]: I wasn't able to get more peers, but I got at least 81 :) https://ipfs.io/ipfs/QmZV7iGTu91w1YePo1EG98siCPxPFrw9Uzf7wAmARusET5

**Comment by eminence** [comment]
look at those happy hackers!  with matching shirts!

**Comment by dignifiedquire** [comment]
giphy-4
